public/app/features/alerting/unified/components/rules/deleted-rules/DeletedRules.tsx

```diff
@@ -0,0 +1,175 @@
+import { useState } from 'react';
+
+import { dateTimeFormat, dateTimeFormatTimeAgo } from '@grafana/data';
+import { Alert, Button, Column, InteractiveTable, LoadingPlaceholder, Stack } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
+import { GrafanaRuleDefinition, RulerRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { trackDeletedRuleRestoreFail, trackDeletedRuleRestoreSuccess } from '../../../Analytics';
+import { alertRuleApi } from '../../../api/alertRuleApi';
+import { GRAFANA_RULER_CONFIG } from '../../../api/featureDiscoveryApi';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { rulerRuleType } from '../../../utils/rules';
+import { UpdatedByUser } from '../../rule-viewer/tabs/version-history/UpdatedBy';
+
+import { ConfirmRestoreDeletedRuleModal } from './ConfirmRestoreDeletedRuleModal';
+
+export const DELETED_RULES_PAGE_SIZE = 30;
+
+export function DeletedRules() {
+  const [confirmRestore, setConfirmRestore] = useState(false);
+  const [restoreRule, setRestoreRule] = useState<RulerRuleDTO | undefined>();
+
+  const {
+    currentData = [],
+    isLoading,
+    error,
+  } = alertRuleApi.endpoints.getDeletedRules.useQuery({
+    rulerConfig: GRAFANA_RULER_CONFIG,
+    filter: {}, // todo: add filters, and limit?????
+  });
+
+  const unknown = t('alerting.deletedRules.unknown', 'Unknown');
+
+  const values = Object.values(currentData);
+  const deletedRules = values.length > 0 ? values[0][0]?.rules : [];
+
+  const showConfirmation = (id: string) => {
+    const ruleTorestore = deletedRules.find(
+      (rule) => rulerRuleType.grafana.rule(rule) && getRowId(rule.grafana_alert) === id
```

Comment:
This endpoint only works for GMA, should we change it's response type to GMA ruler rules?
We wouldn't need to add these typechecks everywhere

Reply:
💯